pkg/action/upgrade.go

```diff
@@ -230,9 +241,10 @@ func (u *UpgradeAction) Run() (err error) {
 		return err
 	}
 
-	// If not upgrading recovery, backup active into passive
-	if !u.spec.RecoveryUpgrade {
-		//TODO this step could be part of elemental package
+	// If not upgrading recovery and booting from non passive, backup active into passive
+	// We dont want to overwrite passive if we are booting from passive as it could mean that active is broken and we would
+	// be overriding a working passive with a broken/unknown  active
+	if u.spec.RecoveryUpgrade == false && bootedFrom != state.Passive {
```

Comment:
Is there a way to test this?

Reply:
I mean in the test suite (not manually)

Reply:
in the end2end or in the unit tests? Unit tests its easy, will add once I see this passing with the new sdk!

end2end may be more complicated as its a whole new test when booting from passive..

Reply:
Fixed the test. As there was no command mock for copying the active into passive, the command was doing nothing, so on master it directly tried to copy it.


Now if the upgrade tries to copy active into passive the test will directly panic

Reply:
tested the test on master which doesnt have the newer workflow and it panics as expected. With this patch it never tries to override the  passive.